src/main/java/de/dennisguse/opentracks/adapters/SensorsAdapterInterface.java

```diff
@@ -0,0 +1,8 @@
+package de.dennisguse.opentracks.adapters;
+
+public interface SensorsAdapterInterface {
```

Comment:
Converting SensorData to abstract was my first thought. But `SensorDataCycling.CadenceAndSpeed` become difficult this approach. Some clue about it ¿?

Reply:
Don't know (yet). I just realized that for the distance computation (i.e., getting it from a sensor), we need to have something like a AggregatedSensorDataSet or so. The speed sensor will provide two values: covered distance since last reset (i.e., trackpoint inserted into db) and the speed (as covered distance by time since last sensor measurement).
So, it might have two values...:man_shrugging:

Reply:
So, we can have sensors that handle two values and other ones that handle one value, right?

Should we create a sensor classes inheritance like the follow?

SensorData
    |___ SimpleSensorData
    |___ ComplexSensorData

In that case, this PR can use the abstract SimpleSensorData (heart rate, power and cadence).

Reply:
Forget what I said in my last comment. I think the best option is what you can review in the second commit just pushed.